File: python_practice/9.回文数.py
#
# @lc app=leetcode.cn id=9 lang=python3
#
# [9] 回文数
#

# @lc code=start
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            str_x = "-" + str(x)
        else:
            str_x = str(x)
        reversed_x = str_x[::-1]
        if reversed_x == str_x:
            return True
        else:
            return False     
        
# @lc code=end

File: cpp/9.回文数.cpp
/*
 * @lc app=leetcode.cn id=9 lang=cpp
 *
 * [9] 回文数
 */
#include<string>

// @lc code=start
class Solution {
public:
    bool isPalindrome(int x) {
        if (x<0) {
            return false;
        }
        std::string x_str = std::to_string(x);
        int left = 0;
        int right = x_str.size();
        for (; left <= right; left++, right--) {
            if (x_str[left] != x_str[right -1]) {
                return false;
            }
        }
        return true;
    }
};
// @lc code=end

